core/templates/dev/head/domain/suggestion/SuggestionThreadObjectFactorySpec.ts

```diff
@@ -80,45 +76,79 @@ describe('Suggestion thread object factory', function() {
     expect(suggestionThread.threadId).toEqual('exploration.exp1.thread1');
-    expect(suggestionThread.suggestion.suggestionType).toEqual(
+
+    var suggestion = suggestionThread.getSuggestion();
+    expect(suggestion.suggestionId).toEqual('exploration.exp1.thread1');
+    expect(suggestion.suggestionType).toEqual(
       'edit_exploration_state_content');
-    expect(suggestionThread.suggestion.targetType).toEqual('exploration');
-    expect(suggestionThread.suggestion.targetId).toEqual('exp1');
-    expect(suggestionThread.suggestion.suggestionId).toEqual(
-      'exploration.exp1.thread1');
-    expect(suggestionThread.suggestion.status).toEqual('accepted');
-    expect(suggestionThread.suggestion.authorName).toEqual('author');
-    expect(suggestionThread.suggestion.newValue.html).toEqual(
-      'new suggestion content');
-    expect(suggestionThread.suggestion.oldValue.html).toEqual(
-      'old suggestion content');
-    expect(suggestionThread.suggestion.lastUpdated).toEqual(1000);
-    expect(suggestionThread.suggestion.getThreadId()).toEqual(
-      'exploration.exp1.thread1');
+    expect(suggestion.targetType).toEqual('exploration');
+    expect(suggestion.targetId).toEqual('exp1');
+    expect(suggestion.status).toEqual('accepted');
+    expect(suggestion.authorName).toEqual('author');
+    expect(suggestion.newValue.html).toEqual('new suggestion content');
+    expect(suggestion.oldValue.html).toEqual('old suggestion content');
+    expect(suggestion.lastUpdated).toEqual(1000);
+    expect(suggestion.getThreadId()).toEqual('exploration.exp1.thread1');
     expect(suggestionThread.isSuggestionThread()).toEqual(true);
     expect(suggestionThread.isSuggestionHandled()).toEqual(true);
-    suggestionThread.suggestion.status = 'review';
+
+    suggestionThread.setSuggestionStatus('review');
     expect(suggestionThread.isSuggestionHandled()).toEqual(false);
     expect(suggestionThread.getSuggestionStatus()).toEqual('review');
     expect(suggestionThread.getSuggestionStateName()).toEqual('state_1');
     expect(suggestionThread.getReplacementHtmlFromSuggestion()).toEqual(
       'new suggestion content');
+  });
+
+  it('should create a new suggestion thread.', function() {
+    var suggestionThread = SuggestionThreadObjectFactory.createFromBackendDicts(
+      suggestionThreadBackendDict, suggestionBackendDict);
+
+    expect(suggestionThread.lastUpdated).toEqual(1000);
+    expect(suggestionThread.originalAuthorName).toEqual('author');
+    expect(suggestionThread.status).toEqual('accepted');
+    expect(suggestionThread.subject).toEqual('sample subject');
+    expect(suggestionThread.summary).toEqual('sample summary');
+    expect(suggestionThread.messageCount).toEqual(10);
+    expect(suggestionThread.threadId).toEqual('exploration.exp1.thread1');
+
+    var suggestion = suggestionThread.getSuggestion();
+    expect(suggestion).toBeUndefined();
+    suggestionThread.setSuggestionStatus();
```

Comment:
@marianazangrossi, are we expecting to pass a status param here? 

@nithusha21, are we expecting the object to throw an error if we set the status of the non-existing suggestion?